models/siegert_neuron.cpp

```diff
@@ -0,0 +1,479 @@
+#include "integerdatum.h"
+
+struct my_params
+{
+  double a;
+  double b;
+};
+
+/* ----------------------------------------------------------------
+ * Siegert implementation
+ * ---------------------------------------------------------------- */
+
+double
+integrand1( double x, void* p )
+{
+  struct my_params* params = ( struct my_params* ) p;
+  double y_th = ( params->a );
+  double y_r = ( params->b );
+  if ( x == 0 )
+  {
+    return exp( -y_th * y_th ) * 2 * ( y_th - y_r );
+  }
+  else
+  {
+    return exp( -( x - y_th ) * ( x - y_th ) )
+      * ( 1. - exp( 2 * ( y_r - y_th ) * x ) ) / x;
+  }
+}
+
+double
+integrand2( double x, void* p )
+{
+  struct my_params* params = ( struct my_params* ) p;
+  double y_th = ( params->a );
+  double y_r = ( params->b );
+  if ( x == 0 )
+  {
+    return 2 * ( y_th - y_r );
+  }
+  else
+  {
+    return ( exp( 2 * y_th * x - x * x ) - exp( 2 * y_r * x - x * x ) ) / x;
+  }
+}
+
+double
+siegert1( double tau_m,
+  double tau_r,
+  double theta,
+  double V_r,
+  double mu,
+  double sigma )
+{
+  double y_th;
+  y_th = ( theta - mu ) / sigma;
+  double y_r;
+  y_r = ( V_r - mu ) / sigma;
+
+  gsl_integration_workspace* w = gsl_integration_workspace_alloc( 1000 );
+
+  double result, error;
+
+  gsl_function F;
+  F.function = &integrand1;
+
+  struct my_params alpha = { y_th, y_r };
+  F.params = &alpha;
+
+  double lower_bound = y_th;
+  double err = 1.;
+  while ( err >= 1e-12 )
+  {
+    err = integrand1( lower_bound, &alpha );
+    if ( err > 1e-12 )
+    {
+      lower_bound /= 2.;
+    }
+  }
+
+  double upper_bound = y_th;
+  err = 1.;
+  while ( err >= 1e-12 )
+  {
+    err = integrand1( upper_bound, &alpha );
+    if ( err > 1e-12 )
+    {
+      upper_bound *= 2.;
+    }
+  }
+
+  gsl_integration_qags(
+    &F, lower_bound, upper_bound, 0.1, 0.1, 1000, w, &result, &error );
+
+  gsl_integration_workspace_free( w );
+
+  return 1. / ( tau_r + exp( y_th * y_th ) * result * tau_m );
+}
+
+double
+siegert2( double tau_m,
+  double tau_r,
+  double theta,
+  double V_r,
+  double mu,
+  double sigma )
+{
+  double y_th;
+  y_th = ( theta - mu ) / sigma;
+  double y_r;
+  y_r = ( V_r - mu ) / sigma;
+
+  gsl_integration_workspace* w = gsl_integration_workspace_alloc( 1000 );
+
+  double result, error;
+
+  gsl_function F;
+  F.function = &integrand2;
+
+  struct my_params alpha = { y_th, y_r };
+  F.params = &alpha;
+
+  double lower_bound = 0.;
+  double upper_bound = 1.;
+  double err = 1.;
+  while ( err >= 1e-12 )
+  {
+    err = integrand2( upper_bound, &alpha );
+    if ( err > 1e-12 )
+    {
+      upper_bound *= 2.;
+    }
+  }
+
+  gsl_integration_qags(
+    &F, lower_bound, upper_bound, 0.1, 0.1, 1000, w, &result, &error );
+
+  gsl_integration_workspace_free( w );
+
+  return 1. / ( tau_r + result * tau_m );
+}
+
+double
+siegert( double tau_m,
+  double tau_syn,
+  double tau_r,
+  double theta,
+  double V_r,
+  double mu,
+  double sigma )
+{
+  double alpha = 2.0652531522312172;
```

Comment:
I assume this is the evaluation of the Riemann zeta function. Can you add a line comment explaining this briefly?